packages/ui/src/components/timeline/Label.tsx

```diff
@@ -48,15 +29,31 @@ export function Label({event, scene}: LabelProps) {
             await findAndOpenFirstUserFile(event.stack);
           }
         }}
-        onMouseDown={e => {
+        onPointerDown={e => {
+          e.currentTarget.setPointerCapture(e.pointerId);
           if (e.button === 1) {
             e.preventDefault();
             player.requestSeek(
               scene.firstFrame +
                 player.status.secondsToFrames(event.initialTime + event.offset),
             );
-          } else {
-            handleDrag(e);
+          }
+        }}
+        onPointerMove={e => {
+          if (e.currentTarget.hasPointerCapture(e.pointerId)) {
+            labelClipDraggingLeftSignal.value = eventTime;
+            setEventTime(
+              eventTime +
+                player.status.framesToSeconds(pixelsToFrames(e.movementX)),
+            );
```

Comment:
One final thing, notice how you assign the `eventTime` here but then in the next line its value is updated to account for the drag. This causes the gray line to be delayed by one frame.
You could improve that by assigning the newly calculated value to the signal, as such
```suggestion
            const newTime = eventTime +
                player.status.framesToSeconds(pixelsToFrames(e.movementX));
            labelClipDraggingLeftSignal.value = newTime;
            setEventTime(newTime);
```